test/client/http_service.test.js

```diff
@@ -0,0 +1,265 @@
+          name: 'rule1',
+          actions: ['get', 'post', 'put', 'patch', 'delete', 'head'],
+          when: {}
+        });
+      service = th.stitchClient.service(SERVICE_TYPE, SERVICE_NAME);
+    });
+
+    afterEach(async() => th.cleanup());
+
+    describe('Submitting a GET request', () => {
+      const requestHandler = (req, res) => {
+        res.writeHead(200, { 'Content-Type': 'text/plan' });
+        res.end('POW!');
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.get(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(200);
+        expect(body.toString()).toBe('POW!');
+      });
+
+      it('with options should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.get({ url: TEST_SERVER_URL });
+
+        expect(statusCode).toBe(200);
+        expect(body.toString()).toBe('POW!');
+      });
+    });
+
+    describe('Submitting a POST request', () => {
+      const requestHandler = async(req, res) => {
+        const body = await readRequestBody(req);
+        res.writeHead(201, { 'Content-Type': 'text/plan' });
+        res.end(body);
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.post(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe('');
+      });
+
+      it('with options should work (raw body)', async() => {
+        handleRequest = requestHandler;
+
+        const requestBody = { foo: 'bar' };
+        const { statusCode, body } = await service.post({
+          url: TEST_SERVER_URL,
+          body: requestBody,
+          encodeBodyAsJSON: true
+        });
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe(JSON.stringify(requestBody));
+      });
+
+      it('with options should work (stringified body)', async() => {
+        handleRequest = requestHandler;
+
+        const requestBody = JSON.stringify({ foo: 'bar' });
+        const { statusCode, body } = await service.post({
+          url: TEST_SERVER_URL,
+          body: requestBody
+        });
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe(requestBody);
+      });
+    });
+
+    describe('Submitting a PUT request', () => {
+      const requestHandler = async(req, res) => {
+        const body = await readRequestBody(req);
+        res.writeHead(204, { 'Content-Type': 'text/plan' });
+        res.end(body);
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.put(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(204);
+        expect(body).toBeFalsy();
+      });
+
+      it('with options should work (raw body)', async() => {
+        handleRequest = requestHandler;
```

Comment:
do we need to do `handleRequest = requestHandler` in every function? can we just set it once in a `beforeAll` inside the `describe` block since we're reusing the same one?

Reply:
ah the `beforeAll` was what I was missing... :doh: I was trying to just explicitly set it inside each `describe` block plainly, and ended up stepping on my feet as describes would inevitably end up using other describes' mock handlers.  thanks!